plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreTableOperationsInsertFailure.java

```diff
@@ -82,12 +78,7 @@ protected LocalQueryRunner createQueryRunner()
             @Override
             public synchronized void replaceTable(String databaseName, String tableName, Table newTable, PrincipalPrivileges principalPrivileges)
             {
-                replaceTableAttempts++;
-                if (replaceTableAttempts == (COMMIT_NUM_RETRIES_DEFAULT + 1)) {
-                    // Ensure that on the last retry operation the `replaceTable` does indeed succeed
-                    // and the table points to a new metadata file
-                    super.replaceTable(databaseName, tableName, newTable, principalPrivileges);
```

Comment:
Why does the second commit change the test scaffolding?

Did you mean to amend the first commit instead?

Reply:
This is the 2nd time this question appears in this PR https://github.com/trinodb/trino/pull/14118#issuecomment-1245864736

I'm squashing therefore the commits.